internal/bminventory/inventory.go

```diff
@@ -4487,3 +4495,79 @@ func formatStaticNetwork(staticNetworkConfig []string) string {
 func (b *bareMetalInventory) GetClusterByKubeKey(key types.NamespacedName) (*common.Cluster, error) {
 	return b.clusterApi.GetClusterByKubeKey(key)
 }
+
+func addStaticNetworkConfigToIgnition(cluster *common.Cluster, log logrus.FieldLogger) ([]NMConnectionData, error) {
+	hostsConfig := strings.Split(cluster.ImageInfo.StaticNetworkConfig, "ZZZZZ")
+	log.Infof("Start configuring static network for %d hosts", len(hostsConfig))
+	executer := &executer.CommonExecuter{}
+	filesList := []NMConnectionData{}
+	for _, hostConfig := range hostsConfig {
+		f, err := executer.TempFile("", "host-config")
+		if err != nil {
+			log.WithError(err).Errorf("Failed to create temp file")
+			return nil, err
+		}
+		_, err = f.WriteString(hostConfig)
+		if err != nil {
+			log.WithError(err).Errorf("Failed to write host config to temp file")
+			return nil, err
+		}
+		f.Close()
+		stdout, stderr, retCode := executer.Execute("nmstatectl", "gc", f.Name())
+		if retCode != 0 {
+			log.Errorf("nstatectl failed, errorCode %d, stderr %s: ", retCode, stderr)
+			return nil, fmt.Errorf("nmstatectl failed, errorCode %d, stderr %s: ", retCode, stderr)
+		}
+		err = createNMConnectionFiles(stdout, &filesList, log)
+		if err != nil {
+			log.WithError(err).Errorf("failed to create NM connection files")
+			return nil, err
+		}
+		os.Remove(f.Name())
+	}
+	return filesList, nil
+}
+
+func createNMConnectionFiles(nmstateOutput string, filesList *[]NMConnectionData, log logrus.FieldLogger) error {
+	var hostNMConnections map[string]interface{}
+	err := yaml.Unmarshal([]byte(nmstateOutput), &hostNMConnections)
+	if err != nil {
+		log.WithError(err).Errorf("Failed to unmarshal nmstate output")
+		return err
+	}
+	connectionsList := hostNMConnections["NetworkManager"].([]interface{})
+	for _, connection := range connectionsList {
+		connectionElems := connection.([]interface{})
+		fileName := connectionElems[0].(string)
+		fileContents, err := formatNMConnection(connectionElems[1].(string), log)
+		if err != nil {
+			return err
+		}
+		log.Infof("Adding NMConnection file <%s>", fileName)
+		newFile := NMConnectionData{
+			FilePath:     filepath.Join(TempNMConnectionsDir, fileName),
+			FileContents: base64.StdEncoding.EncodeToString([]byte(fileContents)),
+		}
+		*filesList = append(*filesList, newFile)
+	}
+	return nil
+}
+
+func formatNMConnection(nmConnection string, log logrus.FieldLogger) (string, error) {
+	start := strings.Index(nmConnection, "interface-name=")
```

Comment:
not sure if worth adding another library, but parsing ini format and adding elements to it can be done by https://github.com/go-ini/ini

Reply:
good idea.  in case that is all the parsing we needed to be done, then i tlhink we leave as is, but in case we will need to do more parsing (vlans, bonding), then we should definitely move to the library. added a TODO for that